src/extension.ts

```diff
@@ -1732,8 +1744,40 @@ async function deleteContextKubernetes(explorerNode: explorer.KubernetesObject)
 }
 
 async function useNamespaceKubernetes(explorerNode: explorer.KubernetesObject) {
-    if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
-        refreshExplorer();
+    if (explorerNode) {
+        if (await kubectlUtils.switchNamespace(kubectl, explorerNode.id)) {
+            refreshExplorer();
+        }
+    } else {
+        promptKindName([kuberesources.allKinds.namespace], undefined,
+            {
+                prompt: 'What namespace do you want to use?',
+                placeHolder: 'Enter the namespace to switch to or press enter to select from available list'
+            },
+            (resource) => {
+                if (resource) {
```

Comment:
I think this logic around checking to see if the current namespace === selected namespace is un-necessary. We should just switch to the selected namespace and refresh the explorer regardless of whether or not it's the current namespace. 

You could, additionally, remove the current namespace from the list of namespaces a user may switch to avoid the current namespace === selected namespace condition entirely!

Reply:
I wish to avoid unnecessary command when current == switch, the time and delay is trivial in local minikube or local kubernetes cluster, but if you are connected to remote cluster this will cause unnecessary delay. I wish to keep to avoid extra command- TBH thats transparent to the user.